### PR TITLE
added support for x-ms-examples in the schema

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

File: schema/swagger-extensions.json [hunk 1]
```diff
@@ -1589,6 +1592,51 @@
...
+        "parameters": {
+          "type": "object",
+          "description": "A dictionary of parameter names and their sample values for a particular operation.",
+          "additionalProperties": {
+            "type": ["object", "number", "array", "integer", "string", "boolean", "null"]
+          }
+        },
+        "responses":{
+          "type": "object",
+          "description": "A dictionary of {possible HTTP status codes (as the dictionary key)} that can be recieved from that operation and {examples (as the dictionary value)} of response {body} and {headers (for long running operations)}.",
```
Comment (TimLovellSmith) [inline schema/swagger-extensions.json]: typo, should be 'received'

File: schema/swagger-extensions.json [hunk 2]
```diff
@@ -1589,6 +1592,51 @@
...
       },
       "uniqueItems": true
+    },
+    "xmsResponseExample": {
+      "type": "object",
+      "description": "Provides the format of specifying an example response in the x-ms-examples extension.",
+      "properties": {
+        "headers": {
+          "type": "object",
+          "description": "Examples of response headers like 'location', 'azure-asyncoperation', 'retry-after', etc.",
```
Comment (TimLovellSmith) [inline schema/swagger-extensions.json]: >'location', 'azure-asyncoperation', 'retry-after', etc.", 

Examples should be in the canonical HTTP casing: 'Location', 'Azure-AsyncOperation', 'Retry-After'

File: schema/swagger-extensions.json [hunk 3]
```diff
@@ -1589,6 +1592,51 @@
...
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "body":{
+          "type": ["object", "number", "array", "integer", "string", "boolean", "null"],
```
Comment (TimLovellSmith) [inline schema/swagger-extensions.json]: Is specifying the type necessary? It seems to not actually be restricting the type in practice.

Comment (TimLovellSmith) [inline schema/swagger-extensions.json]: Could perhaps use further description - If 'null' is specified or 'string' is specified, I wonder how they are interpreted? An empty body, or a body containing the text 'null' (which I have seen some servers return :p) ? Would a string value be assumed to be plain text of the body?

File: schema/swagger-extensions.json [hunk 4]
```diff
@@ -1589,6 +1592,51 @@
...
+          "description": "A dictionary of parameter names and their sample values for a particular operation.",
+          "additionalProperties": {
+            "type": ["object", "number", "array", "integer", "string", "boolean", "null"]
+          }
+        },
+        "responses":{
+          "type": "object",
+          "description": "A dictionary of {possible HTTP status codes (as the dictionary key)} that can be recieved from that operation and {examples (as the dictionary value)} of response {body} and {headers (for long running operations)}.",
+          "additionalProperties": {
+            "$ref": "#/definitions/xmsResponseExample"
```
Comment (veronicagg) [inline schema/swagger-extensions.json]: should the schema of the response be more explicit with patternProperties (for the status codes) and object for their values?

File: schema/swagger-extensions.json (hunk 3 above)
Comment (veronicagg) [inline schema/swagger-extensions.json]: should body be of type object?

File: schema/swagger-extensions.json (hunk 4 above)
Comment (veronicagg) [inline schema/swagger-extensions.json]: where's status code of the response currently being modeled?

File: schema/swagger-extensions.json [hunk 5]
```diff
@@ -1589,6 +1592,51 @@
...
+    },
+    "xmsExampleObject": {
+      "type": "object",
+      "required": ["parameters", "responses"],
+      "description": "Describes the format of an example defined in the x-ms-examples extension.",
+      "properties": {
+        "parameters": {
+          "type": "object",
+          "description": "A dictionary of parameter names and their sample values for a particular operation.",
+          "additionalProperties": {
```
Comment (amarzavery) [inline schema/swagger-extensions.json]: The schema is a static thing. I do not know what kind of parameters would one want to provide. I mean, we surely know some common one's like subscriptionId, api-version, resourceGroupName. But these common one's are not always present as well. AD Graph does not have resourceGroupName and subscriptionId. 

We can go either way:
1 - The current approach that is taken in the PR
2 - Model the commonly used parameters as optional properties and set additionalProperties to true (so people can add more parameters).

What do you think?

**Comment by TimLovellSmith** [comment]
:shipit:

**Pull request merged**